flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/ContextResolvedTable.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    public static ContextResolvedTable temporary(
+            ObjectIdentifier identifier, ResolvedCatalogBaseTable<?> resolvedTable) {
+        return new ContextResolvedTable(identifier, null, resolvedTable, false);
+    }
+
+    public static ContextResolvedTable anonymous(ResolvedCatalogBaseTable<?> resolvedTable) {
+        return new ContextResolvedTable(
+                ObjectIdentifier.ofAnonymous(
+                        generateAnonymousStringIdentifier(null, resolvedTable)),
+                null,
+                resolvedTable,
+                true);
+    }
+
+    public static ContextResolvedTable anonymous(
+            String hint, ResolvedCatalogBaseTable<?> resolvedTable) {
+        return new ContextResolvedTable(
+                ObjectIdentifier.ofAnonymous(
+                        generateAnonymousStringIdentifier(hint, resolvedTable)),
+                null,
+                resolvedTable,
+                true);
+    }
+
+    private ContextResolvedTable(
+            ObjectIdentifier objectIdentifier,
+            @Nullable Catalog catalog,
+            ResolvedCatalogBaseTable<?> resolvedTable,
+            boolean anonymous) {
+        this.objectIdentifier = Preconditions.checkNotNull(objectIdentifier);
+        this.catalog = catalog;
+        this.resolvedTable = Preconditions.checkNotNull(resolvedTable);
+        this.anonymous = anonymous;
+    }
+
+    public boolean isAnonymous() {
+        return this.anonymous;
+    }
+
+    /** @return true if the table is temporary. An anonymous table is always temporary. */
+    public boolean isTemporary() {
+        return catalog == null;
+    }
+
+    public boolean isPermanent() {
+        return !isTemporary();
+    }
+
+    public ObjectIdentifier getIdentifier() {
+        return objectIdentifier;
+    }
+
+    /** Returns empty if {@link #isPermanent()} is false. */
+    public Optional<Catalog> getCatalog() {
+        return Optional.ofNullable(catalog);
+    }
+
+    /** Returns a fully resolved catalog object. */
+    @SuppressWarnings("unchecked")
+    public <T extends ResolvedCatalogBaseTable<?>> T getResolvedTable() {
+        return (T) resolvedTable;
+    }
+
+    public ResolvedSchema getResolvedSchema() {
+        return resolvedTable.getResolvedSchema();
+    }
+
+    /** Returns the original metadata object returned by the catalog. */
+    @SuppressWarnings("unchecked")
+    public <T extends CatalogBaseTable> T getTable() {
+        return (T) resolvedTable.getOrigin();
+    }
+
+    /**
+     * Copy the {@link ContextResolvedTable}, replacing the underlying {@link CatalogTable} options.
+     */
+    public ContextResolvedTable copy(Map<String, String> newOptions) {
+        if (resolvedTable.getTableKind() == CatalogBaseTable.TableKind.VIEW) {
+            throw new ValidationException(
+                    String.format("The view '%s' cannot be enriched with new options.", this));
```

Comment:
"View '%s' cannot be enriched with new options. Hints can only be applied to tables."

Reply:
Added this specific error about hints at the call site.